.github/workflows/main.yml

```diff
@@ -0,0 +1,97 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Pyccel tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  Linux:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.5]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        uses: ./.github/actions/linux_install
+      - name: Install python dependencies
+        uses: ./.github/actions/pip_installation
+      - name: Coverage install
+        uses: ./.github/actions/coverage_install
+      - name: Test with pytest
+        uses: ./.github/actions/pytest_run
+      - name: Collect coverage information
+        continue-on-error: True
+        uses: ./.github/actions/coverage_collection
+
+  Linux-Master:
+
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```

Comment:
Can the `if` condition be put in the strategy matrix?

Reply:
Sorry, probably not... I don't think they support sparse matrices! 😅

Reply:
You can put an `includes` or `excludes` s section but for some reason you can't put an if in that